plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StdLibConverter.kt

```diff
@@ -17,69 +16,192 @@ fun VariableEmbedding.sameSize(): ExpEmbedding =
 fun VariableEmbedding.increasedSize(amount: Int): ExpEmbedding =
     EqCmp(FieldAccess(this, ListSizeFieldEmbedding), Add(Old(FieldAccess(this, ListSizeFieldEmbedding)), IntLit(amount)))
 
-fun NamedFunctionSignature.stdLibPreConditions(): List<ExpEmbedding> =
-    NameMatcher.match(this.name) {
-        ifInCollectionsPkg {
-            ifFunctionName("get") {
-                val receiver = receiver!!
-                val indexArg = formalArgs[1]
-                return listOf(
-                    GeCmp(
-                        indexArg,
-                        IntLit(0),
-                        SourceRole.ListElementAccessCheck(SourceRole.ListElementAccessCheck.AccessCheckType.LESS_THAN_ZERO)
-                    ),
-                    GtCmp(
-                        FieldAccess(receiver, ListSizeFieldEmbedding),
-                        indexArg,
-                        SourceRole.ListElementAccessCheck(SourceRole.ListElementAccessCheck.AccessCheckType.GREATER_THAN_LIST_SIZE)
-                    ),
-                )
+sealed interface StdLibReceiverInterface {
+    fun match(function: NamedFunctionSignature): Boolean
+}
+
+sealed interface PresentInterface : StdLibReceiverInterface {
+    val interfaceName: String
+    override fun match(function: NamedFunctionSignature): Boolean =
+        function.receiverType?.isInheritorOfCollectionTypeNamed(interfaceName) == true
```

Comment:
nit: `b == true` is the same as just `b`

Reply:
Well, the left part of comparison here has `Boolean?` type. But may be `?: false` is clearer indeed.

Reply:
Ah, I see.  Yeah, I tihnk `?: false` would be better